tests/components/opentherm_gw/test_config_flow.py

```diff
@@ -189,21 +196,45 @@ async def test_options_form(hass):
 
     result = await hass.config_entries.options.async_configure(
```

Comment:
Make sure to set up the integration before starting the options flow in the beginning of the test. Use `hass.config_entries.async_setup`. This is needed to make sure that the config flow module is loaded if the test is run in isolation.

Reply:
Should there also be `with patch() ` in this test form? Or is it enough to place `hass.config_entries.async_setup(entry.entry_id)` and `await hass.async_block_till_done()` under `entry.add_to_hass(hass)`?

Reply:
We need to patch something so that there's no I/O in the test. If we only care about the isolated options flow, we could patch the integration setup function `async_setup_entry`.

Reply:
Yes, exactly!

Reply:
Is it correct to end te test with 
```
assert len(mock_setup.mock_calls) == 0
assert len(mock_setup_entry.mock_calls) == 0
```
Local test will fail if I assert 1
Am I doing something wrong or is this expected?

Reply:
So might just go without patch.

If I place
```
hass.config_entries.async_setup(entry.entry_id)
await hass.async_block_till_done()
```
and then run the test I only get this `RuntimeWarning: coroutine 'ConfigEntries.async_setup' was never awaited`
Is it ok to leave it at that? Or remove `await hass.async_block_till_done()` as well?

Reply:
We don't want to go for "as little warnings as possible", but rather for "no warnings whatsoever" :wink: 
Try `await hass.config_entries.async_setup(entry.entry_id)` instead.

Reply:
If i do that i'm getting `TypeError: async_block_till_done() takes 1 positional argument but 2 were given`

Reply:
Then you'll have a call to `hass.async_block_till_done()` with an argument somewhere, the test log will probably tell you where.
Rather than troubleshooting error messages one at a time, you can post what you have so far (the full test file, use hastebin or a similar service) and I will see if I can help out with your issues.